File: src/main/java/com/vjiki/music/service/ChatService.java
package com.vjiki.music.service;

import java.util.List;
import java.util.UUID;

import com.vjiki.music.dto.ChatResponse;
import com.vjiki.music.dto.CreateChatRequest;

public interface ChatService {

    List<ChatResponse> getChatsByUserId(UUID userId);

    ChatResponse getChatById(UUID chatId);

    ChatResponse createChat(CreateChatRequest request);

    void deleteChat(UUID chatId);
}

File: src/main/kotlin/com/vjiki/music/service/ChatService.kt
package com.vjiki.music.service

import com.vjiki.music.dto.ChatResponse
import com.vjiki.music.dto.CreateChatRequest
import java.util.UUID

interface ChatService {
    fun getChatsByUserId(userId: UUID): List<ChatResponse>
    fun getChatById(chatId: UUID): ChatResponse
    fun createChat(request: CreateChatRequest): ChatResponse
    fun deleteChat(chatId: UUID)
}

